src/components/questionStructure/Task/variants/HandwritingTask/ViewOnlyCanvas.tsx

```diff
@@ -1,30 +1,41 @@
 import { Excalidraw } from '@excalidraw/excalidraw'
-import {
-  ExcalidrawImperativeAPI,
-  ExcalidrawInitialDataState,
-} from '@excalidraw/excalidraw/types/types'
+import { ExcalidrawElement } from '@excalidraw/excalidraw/types/element/types'
+import { ExcalidrawImperativeAPI } from '@excalidraw/excalidraw/types/types'
 import { Box, Card } from '@radix-ui/themes'
 import React, { useEffect, useState } from 'react'
 
 interface ViewOnlyCanvasProps {
-  initialData: ExcalidrawInitialDataState
+  initialData: readonly ExcalidrawElement[]
 }
 
 export const ViewOnlyCanvas: React.FC<ViewOnlyCanvasProps> = ({ initialData }) => {
   const [excalidrawAPI, setExcalidrawAPI] = useState<ExcalidrawImperativeAPI | null>(null)
 
   useEffect(() => {
+    if (excalidrawAPI && initialData) {
+      excalidrawAPI.updateScene({
+        elements: initialData,
+      })
+    }
     setTimeout(() => {
       excalidrawAPI?.scrollToContent(undefined, {
         fitToContent: true,
       })
     })
-  }, [excalidrawAPI])
+  }, [excalidrawAPI, initialData])
 
-  return initialData && initialData.elements?.length ? (
+  return initialData && initialData?.length ? (
     <Card>
-      <Box className="excalidraw-view-container excalidraw-box">
-        <Excalidraw excalidrawAPI={setExcalidrawAPI} viewModeEnabled initialData={initialData} />
+      <Box
+        className="excalidraw-view-container excalidraw-box"
+        height="calc(100% - 4px)"
+        style={{ margin: '2px' }}
```

Comment:
Are these styles still relevant? I believe they should be already part of `excalidraw-view-container`

Reply:
You're right. This has been addressed in eb0c209ea5f973a7c6d9a420e562347fa8d1572a.